lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
```

Comment:
The payload needs to have a flag indicating whether it expects to receive a UUID and then handle it if so. Otherwise you need an entirely new handler.